openhands/server/config/server_config.py

```diff
@@ -27,30 +27,32 @@ class ServerConfig(ServerConfigInterface):
         'openhands.server.user_auth.default_user_auth.DefaultUserAuth'
     )
 
-    def verify_config(self):
+    def verify_config(self) -> None:
         if self.config_cls:
             raise ValueError('Unexpected config path provided')
 
-    def get_config(self):
-        config = {
-            'APP_MODE': self.app_mode,
+    async def get_config(self) -> dict[str, str]:
+        config: dict[str, str] = {
+            'APP_MODE': str(self.app_mode),
```

Comment:
We want to pass the enum here; the rest of the app compares against AppMode enum rather than the string

```suggestion
            'APP_MODE': self.app_mode,
```